clar2wasm/src/words/equal.rs

```diff
@@ -0,0 +1,85 @@
+use crate::wasm_generator::{clar2wasm_ty, ArgumentsExt, GeneratorError, WasmGenerator};
+use clarity::vm::{types::TypeSignature, ClarityName, SymbolicExpression};
+use walrus::ir::BinaryOp;
+
+use super::Word;
+
+#[derive(Debug)]
+pub struct IsEq;
+
+impl Word for IsEq {
+    fn name(&self) -> ClarityName {
+        "is-eq".into()
+    }
+
+    fn traverse(
+        &self,
+        generator: &mut WasmGenerator,
+        builder: &mut walrus::InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        args: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        // Traverse the first operand pushing it onto the stack
+        let first_op = args.get_expr(0)?;
+        generator.traverse_expr(builder, first_op)?;
+
+        // Save the first_op to a local to be further used.
+        // This allows to use the firt_op value without
+        // traversing again the expression.
+        let ty = generator
+            .get_expr_type(first_op)
+            .expect("is-eq value expression must be typed")
+            .clone();
+        let wasm_types = clar2wasm_ty(&ty);
+        let mut val_locals = Vec::with_capacity(wasm_types.len());
+        for local_ty in wasm_types.iter().rev() {
+            let local = generator.module.locals.add(*local_ty);
+            val_locals.push(local);
+            builder.local_set(local);
+        }
+        val_locals.reverse();
+
+        // Traverse the second operand pushing it onto the stack.
+        let sec_op = args.get_expr(1)?;
+        generator.traverse_expr(builder, sec_op)?;
```

Comment:
You cannot do that here, otherwise this valid contract function cannot work:
```
(define-public (one-argument-equal)
    (ok (is-eq 42))
)
```
Please add this function to "equal.clar" for the tests (answer should be `true`).

Reply:
Done